node/sc/bridge_manager.go

```diff
@@ -426,21 +439,40 @@ func (bi *BridgeInfo) GetCurrentBlockNumber() (uint64, error) {
 
 // DecodeRLP decodes the Klaytn
 func (b *BridgeJournal) DecodeRLP(s *rlp.Stream) error {
-	var elem struct {
+	var LegacyBridgeAddrInfo struct {
 		LocalAddress  common.Address
 		RemoteAddress common.Address
 		Paired        bool
 	}
-	if err := s.Decode(&elem); err != nil {
-		return err
+	var BridgeAddrInfo struct {
+		BridgeAlias   string
+		LocalAddress  common.Address
+		RemoteAddress common.Address
+		Paired        bool
+	}
+	if !b.isLegacyBridgeJournal {
+		if err := s.Decode(&BridgeAddrInfo); err != nil {
+			logger.Trace("Failed to decode. Try decode again with legacy structure")
+			b.isLegacyBridgeJournal = true
```

Comment:
I have 2 question in this code.
1. Is the reason of decoding failure always different BridgeJournal struct? 
2. If decode fails, then it tries to decode automatically?

Reply:
1. Yes. There's only one case, that is the legacy bridge journal (no alias field). If the decode is failed when a struct is neither legacy journal nor aliased journal, no journal is loaded.
2. Yes. Please [see](https://github.com/klaytn/klaytn/pull/1438/files/328d7f2d2e54ab6f6b8acb3cf695e04d10019024#diff-5354626cdfa967dd6b4dbe6fd55e43f5bc198018bb25bb38fc6d21b18fab9f2eR91-R118).